roles/chameleon_prometheus/templates/snmp_generator.yml.j2

```diff
@@ -5,10 +5,24 @@ modules:
   {% if switch.device_type not in device_types %}
   {{ device_types.append(switch.device_type) }}
   {{ switch.device_type }}:
-    walk:
-      - interfaces
-    version: 1
+    version: 2
     auth:
       community: {{ switch.snmp.community_string }}
+    walk:
+      - system
+      - interfaces
+      - ifXTable
+    lookups:
+      - source_indexes: [ifIndex]
+        lookup: ifName
+    overrides:
```

Comment:
For my own knowledge, what does this do? I don't understand how setting "ignore" to true means "Lookup metric"

Reply:
This is really poorly documented by them, and depends on a bit of SNMP knowledge.

ifIndex is an arbitrary but unique (per switch) identifier associated with an interface.
to look up a value like ifHCInOctets, for a specific interface, you'd query the OID "ifHCInOctets.<ifIndex>".

This is 

What this 'lookups" field does is for each unique ifIndex, get the associated ifName, and add it as a prometheus label.

the overrides / ignore part is stating not to otherwise add those labels. E.g. ONLY add the specific lookup.
This is copied from their example, but is because you could add more specific matching under the lookups, such as a dell specific OID which would imply that ifDescr was correct instead of ifName.